homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,746 @@
+"""Hive Integration - sensor."""
+import logging
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.helpers.entity import Entity
+from homeassistant.loader import get_component
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices,
+                   device_list, discovery_info=None):
+    """Setup Hive sensor devices."""
+    hive_comp = get_component('hive')
+
+    if len(device_list) > 0:
+        for device in device_list:
+            if ("HA_DeviceType" in device and "Hive_NodeID" in device and
+                    "Hive_NodeName" in device):
+                if device["HA_DeviceType"] == "Heating_CurrentTemperature":
+                    add_devices([CurrentTemperature(hass,

```

Comment:
This is too many classes with duplicated code. For each platform, try to consolidate into one class (i.e. HiveSensorEntity, HiveClimateEntity, etc.), and just do branching internally based on HA_DeviceType where necessary.

Reply:
consolidated code as you have indicated